From 4b146a15ac08be10870c3886f9fe681d261de268 Mon Sep 17 00:00:00 2001
From: Xu Bing <bing.xu@intel.com>
Date: Thu, 13 Mar 2025 09:04:50 +0800
Subject: [PATCH] Force to set letterbox type to wallpaper

Default type of letterbox is LETTERBOX_BACKGROUND_SOLID_COLOR, so
letter box is dark, set type to LETTERBOX_BACKGROUND_WALLPAPER and
force to show wallpaper.

Tracked-on: OAM-130945
Signed-off-by: Xu Bing <bing.xu@intel.com>
---
 .../java/com/android/server/wm/LetterboxUiController.java    | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/services/core/java/com/android/server/wm/LetterboxUiController.java b/services/core/java/com/android/server/wm/LetterboxUiController.java
index 735cbc4e4287..aafe1a6b08a7 100644
--- a/services/core/java/com/android/server/wm/LetterboxUiController.java
+++ b/services/core/java/com/android/server/wm/LetterboxUiController.java
@@ -1607,6 +1607,11 @@ final class LetterboxUiController {
                         // Check that blur is supported by a device if blur radius is provided.
                         && (getLetterboxWallpaperBlurRadiusPx() <= 0
                                 || isLetterboxWallpaperBlurSupported());
+        //force set wallpaper when background type is LETTERBOX_BACKGROUND_WALLPAPER
+        if (letterboxBackgroundType == LETTERBOX_BACKGROUND_WALLPAPER) {
+            wallpaperShouldBeShown = true;
+            Slog.w(TAG, "set letterbox Wallpaper to be shown.....");
+        }
         if (mShowWallpaperForLetterboxBackground != wallpaperShouldBeShown) {
             mShowWallpaperForLetterboxBackground = wallpaperShouldBeShown;
             mActivityRecord.requestUpdateWallpaperIfNeeded();
-- 
2.34.1

